src/ui/atoms/text-area/index.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import styled, { StyledComponent } from 'styled-components';
+import { Variant } from 'lib/types';
+
+/**
+ * --woly-font-size
+ * --woly-rounding
+ * --woly-line-height
+ * --woly-background
+ * --woly-background-disabled
+ * --woly-border
+ * --woly-border-focus
+ * --woly-border-disabled
+ * --woly-color
+ * --woly-color-disabled
+ *
+ */
+interface TextAreaProps
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {
+  className?: string;
+  cols?: number;
+  disabled?: boolean;
+  label?: React.ReactNode;
+  maxLength?: number;
+  minlength?: number;
+  name: string;
+  onChange: (e: React.ChangeEvent<HTMLTextAreaElement>) => unknown;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLTextAreaElement>) => unknown;
+  overflow?: string;
+  placeholder?: string;
+  resize?: string;
+  rows?: number;
+  textError?: string;
+  textareaRef?: any;
+  value?: string;
+  wrap?: string;
+}
+
+const TextAreaBase: React.FC<TextAreaProps & Variant> = ({
+  className,
+  cols,
+  disabled,
+  label,
+  maxLength,
+  minlength,
+  name,
+  onChange,
+  onKeyDown,
+  overflow,
+  placeholder,
+  resize,
+  rows,
+  textError,
+  textareaRef,
+  value,
+  variant = 'default',
+  wrap,
+  ...p
+}) => {
+  return (
+    <TextAreaWrapper
+      data-overflow={overflow}
+      data-resize={resize}
+      data-variant={variant}
+     >
+      {label && <span>{label}</span>}
+      <textarea
+        className={className}
+        cols={cols}
+        data-variant={variant}
+        disabled={disabled}
+        maxLength={maxLength}
+        minLength={minlength}
+        name={name}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        placeholder={placeholder}
+        ref={textareaRef}
+        rows={rows}
+        value={value}
+        wrap={wrap}
+        {...p}
+      />
+      <Containrer>
```

Comment:
Тогда и не понадобятся вложенные компоненты.
сможешь их просто помечать:

```suggestion
      <div data-block="container">
```